### PR TITLE
Added basic Installation targets

### PR DESCRIPTION
**Pull request by capnkenny**
CMake Install should now:
-- Copy the required DLL/Lib (Dylib not confirmed atm)
-- Copy all headers (respecting the hierarchy provided)
-- Copy the shaders _always_
-- Generate a `.cmake` to find the proper config

### CODE REVIEW

File: CMakeLists.txt
Comment (RubyNova) [inline CMakeLists.txt:90]: Do we need the double colon here?

**Review by RubyNova — APPROVED** [review]
LGTM but gonna wait on @FiniteReality before merge

File: CMakeLists.txt
Comment (capnkenny) [reply CMakeLists.txt:90]: I believe not necessarily but it may help? I framed this off of Fabulist's export

Comment (RubyNova) [reply CMakeLists.txt:90]: I see, ok, well what do I know 😆

Comment (FiniteReality) [reply CMakeLists.txt:90]: The namespace is what it generates as the prefix in the CMake find_package scripts, for grouping all of the targets from a specific library together. For example:
```cmake
# Create imported target fabulist::lua
add_library(fabulist::lua STATIC IMPORTED)

set_target_properties(fabulist::lua PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "LUA_ANSI"
  INTERFACE_COMPILE_OPTIONS "\$<\$<CXX_COMPILER_ID:MSVC>:/experimental:external>;\$<\$<CXX_COMPILER_ID:MSVC>:/external:I>;\$<\$<CXX_COMPILER_ID:MSVC>:/external:W0>;\$<\$<CXX_COMPILER_ID:MSVC>:/external:templates->"
  INTERFACE_INCLUDE_DIRECTORIES "\$<\$<NOT:\$<CXX_COMPILER_ID:MSVC>>:>"
  INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "\$<\$<NOT:\$<CXX_COMPILER_ID:MSVC>>:>"
)
```

File: src/NovelRT/CMakeLists.txt
Comment (FiniteReality) [inline src/NovelRT/CMakeLists.txt:165]: Post MVP I want to restructure the engine such that each "area" is its own static/object library, and is then linked into the main "engine" - this should simplify the layout of our include files and flatten them so that we can use PUBLIC_HEADER or whatever. Hopefully this'll enforce a lot more separation between various components.

**Review by FiniteReality — APPROVED** [review]
Seems good enough to me

**Pull request merged**
